plat/hisilicon/hikey960/drivers/pwrc/hisi_pwrc.c

```diff
@@ -149,11 +149,17 @@ void hisi_clear_cpuidle_flag(unsigned int cluster, unsigned int core)
 
 int hisi_test_ap_suspend_flag(unsigned int cluster)
 {
-	unsigned int val;
+	unsigned int val1;
+	unsigned int val2;
 
-	val = mmio_read_32(CPUIDLE_FLAG_REG(cluster));
-	val &= AP_SUSPEND_FLAG;
-	return !!val;
+	val1 = mmio_read_32(CPUIDLE_FLAG_REG(0));
+	val1 &= AP_SUSPEND_FLAG;
+
+	val2 = mmio_read_32(CPUIDLE_FLAG_REG(1));
+	val2 &= AP_SUSPEND_FLAG;
+
+	val1 |= val2;
+	return !!val1;
```

Comment:
I personally find the double negation less readable.
perhaps:
return (val1 == 1);

Reply:
sorry, I meant:
return (val1 != 0);

Reply:
Yes, will change to this way (and upload the new patch after settle down the issue reported by Valentine).